File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        vector<int> s;
        int i = 0;
        int j = 0;
        
        while (i < m && j < n){
            if (nums1[i] > nums2[j]){
                s.push_back(nums2[j]);
                j++;
            }
            else{
                s.push_back(nums1[i]);
                i++;
            }   
        }
        
        while (i < m){
            s.push_back(nums1[i]);
            i++;
        }
        while (j < n){
            s.push_back(nums2[j]);
            j++;
        }
        
        for(int i = 0 ; i < m + n; i++){
            nums1[i] = s[i];
        }
    }
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        ptr1 = len(nums1) - len(nums2) - 1
        ptr2 = len(nums2) - 1
        ptr3 = len(nums1) - 1
        
        
        while (ptr1 > -1 and ptr2 > -1) or (ptr1 < 0 and ptr2 > -1):
            if ptr1 > -1 and nums1[ptr1] >= nums2[ptr2]:
                nums1[ptr3] = nums1[ptr1]
                ptr3 -= 1
                ptr1 -= 1
            else:
                nums1[ptr3] = nums2[ptr2]
                ptr3 -= 1
                ptr2 -= 1
        
        
        # while ptr2 > -1:
        #     nums1[ptr3] = nums2[ptr2]
        #     ptr2 -= 1
        #     ptr3 -= 1
        
        
        
                
        